WordPress/gutenberg
# Revert/block support stabilization

## What?

Reverting the changes of this PR https://github.com/WordPress/gutenberg/pull/67552

## Why?

The approach taken to block support stabilization ironically needs some further time to stabilize itself 

## How?
Reverts the commits from the PRs listed above.

## Testing Instructions

1. Add a variety of different blocks that opt into typography supports. Ensure both static and dynamic blocks are covered.
2. Apply some global typography styles for the blocks being tested.
3. Select the individual block instances and apply new typography styles to override global styles. Confirm these are applied correctly in the editor.
4. Save and confirm the typography support styles are correctly applied on the frontend.
5. Back in the editor, cross check the default controls defined in the blocks' block.json with the controls displayed by default in the block inspector.
6. Repeat the process from 1-5, this time checking border block supports function as before.
7. Confirm that blocks that skip the serialization of block support styles and classes still function e.g. Images with borders etc.
